### PR TITLE
Removed extra 'version' field on GET requests and implemented dynamic incrementing on save

### PR DESCRIPTION
**Pull request by Caleb-Rogers**

### CODE REVIEW

**Review by Mike-Volchko — APPROVED** [review]
Looks good.

File: server/controller/apis/goalController.js
Comment (cmarcusdimarco) [inline server/controller/apis/goalController.js:172]: Looks like the new functionality will increment the version number even if the existing goal is saved with no changes. If this is intentional due to thinking that use case will be very rare, I don't mind approving. However, it looks like checking for changes would be as simple as comparing the existingGoal fields with the goal passed in via the req parameter.

**Review by cmarcusdimarco — COMMENTED** [review]
Please see the comment on line 172 - I am happy to approve if this is the intended behavior for a reason I don't see, but we should consider the benefits of checking for changes before incrementing version number.

**Comment by Mike-Volchko** [comment]
Looks like the new changes address @cmarcusdimarco comment. Once this PR is approved by @cmarcusdimarco and @briangormanly we can merge.

**Review by briangormanly — APPROVED** [review]
looks good.

**Review by briangormanly — CHANGES_REQUESTED** [review]
It looks good but there is a conflict now with main.

**Comment by Caleb-Rogers** [comment]
Added a couple of commits that fixed some issues and resolved the merge conflicts which were caused by the change from goal.id to goal.goalId. This PR is now ready to merge.

**Review by briangormanly — APPROVED** [review]
Update to date with main, ready to go.

**Pull request merged**
